free-4.md

```diff
@@ -0,0 +1,128 @@
+# Free project 4
+
+## The user and a language
+This section describes who the project would serve and why a language might be a
+good way to meet their needs.
+
+
+### What's the need?
+_What need is met by your idea? Who are you helping? What is that person's
+experience like now? What would their experience be like if you could help 
+them?_
+
+My final idea is to create a DSL which is designed specifically for plotting
+high dimensional data.  I would be helping scientists, data analysts, and
+anyone who needs to create three dimensional plots.
+
+### Why a language?
+_Why is a DSL appropriate for your user(s)? How does it address the need?_
+
+The problem of plotting high dimensional data is essentially a problem of
+describing the shape of a surface in some dimension and rendering this
+information in a useful manner.  Having an effective way to express the data
+can allow the computer to worry about the rendering rather than leaving
+it up to the user.
+
+### Why you?
+_What excites you about this idea? How did you come up with it?_
+
+I was often annoyed by the use of boilerplate code in matplotlib and
+difficulties in verifying that 3D plots were correct.  I was thinking
+that having a better way to express the data might be the bottleneck.
+For example, in matplotlib, one must create a "meshgrid" and set values
+of a function to corresponding values, rather than simply provide a
+set of (x,y,z) values to plot, or a function and a range of `x`s and `y`s.
```

Comment:
This is the most annoying thing to do. Every single time I want to make a 3d surface plot in matplotlib I always forget this, and I always have to go google how to do it.